salt/_modules/containerd.py

```diff
@@ -0,0 +1,60 @@
+'''
+Various functions to interact with :program:`containerd` using :command:`ctr`.
+'''
+
+import re
+import logging
+
+import salt.utils.path
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'containerd'
+
+
+def __virtual__():
+    if not salt.utils.path.which('ctr'):
+        return (False, '\'ctr\' command not found')
+
+    return __virtualname__
+
+
+_LOAD_RE = re.compile('Loaded image: (?P<name>.*)')
```

Comment:
I don't think you need to compile this regexp.

> Note: The compiled versions of the most recent patterns passed to re.compile() and the module-level matching functions are cached, so programs that use only a few regular expressions at a time needn’t worry about compiling regular expressions.

From https://docs.python.org/3/library/re.html#re.compile

And about the pattern: `.*` accept an empty string, is it what we want?

Reply:
I wouldn't be surprised Salt (and its modules) use regular expressions all over the place, so there's no way for me to 'predict' this one could be cached automagically.

For empty string: sure, could make it slightly more strict, though the tool will likely never output that, and if it does, then the name of the image would be 'empty string' anyway.